app/js/controller/list_controller.js

```diff
@@ -107,7 +107,10 @@ export default class ListController extends Controller {
   handleInstall(data) {
     var manifestURL = data.manifestURL;
     if (this.list[manifestURL].mozApp) {
-      this.list[manifestURL].mozApp.launch();
+      // Addons cannot be launched so do nothing on button click.
+      if (data.type !== 'addon') {
+        this.list[manifestURL].mozApp.launch();
+      }
```

Comment:
It is disabled in this case. For some reason, gaia-buttons still propagates the click event even when it is marked as disabled. I'll file a bug with the component when I have some time to look into it.